GitUI/CommandsDialogs/BrowseDialog/FormOpenDirectory.cs

```diff
@@ -107,8 +120,14 @@ private void folderGoUpbutton_Click(object sender, EventArgs e)
 
         private void _NO_TRANSLATE_Directory_TextChanged(object sender, EventArgs e)
         {
-            DirectoryInfo currentDirectory = new DirectoryInfo(_NO_TRANSLATE_Directory.Text);
-            folderGoUpbutton.Enabled = (currentDirectory.Parent != null);
+            try
+            {
+                DirectoryInfo currentDirectory = new DirectoryInfo(_NO_TRANSLATE_Directory.Text);
+                folderGoUpbutton.Enabled = (currentDirectory.Parent != null);
```

Comment:
1. redundant braces
2. you need to check whether the path provided exists and only enable the button if it is 
https://msdn.microsoft.com/en-us/library/system.io.directoryinfo.directoryinfo(v=vs.110).aspx

> This constructor does not check if a directory exists. This constructor is a placeholder for a string that is used to access the disk in subsequent operations.